src/components/TodoItem.tsx

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onDeleteTodo: (todoId: number) => void;
+  // isCompleted: boolean;
+  // setCompletedTodosId: React.Dispatch<React.SetStateAction<number[]>>;
+  isLoading: boolean;
+};
+
+export const TodoItem: React.FC<Props> = props => {
+  const { todo, onDeleteTodo, isLoading } = props;
+
+  // const handleComplete = (event: React.ChangeEvent<HTMLInputElement>) => {
+  //   if (event.target.checked) {
+  //     setCompletedTodosId(prev => [...prev, todo.id]);
+  //   } else {
+  //     setCompletedTodosId(prevTodosId =>
+  //       prevTodosId.filter(todoId => todoId !== todo.id),
+  //     );
+  //   }
+  // };
+
+  return (
+    <div data-cy="Todo" className={cn('todo', { completed: todo.completed })}>
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          // onChange={event => handleComplete(event)}
+        />
+      </label>
+
+      <span data-cy="TodoTitle" className="todo__title">
+        {todo.title}
+      </span>
+
+      {/* Remove button appears only on hover */}
+      <button
+        type="button"
+        className="todo__remove"
+        data-cy="TodoDelete"
+        onClick={() => onDeleteTodo(todo.id)}
+        disabled={isLoading}
+      >
+        ×
+      </button>
+
+      {/* overlay will cover the todo while it is being deleted or updated */}
+      <div
+        data-cy="TodoLoader"
+        className={cn('modal overlay', {
+          'is-active': isLoading || todo.id === 0,
```

Comment:
```suggestion
          'is-active': isLoading,
```

Reply:
if I remove this part of code it doesn't work correctly